packages/storybook/tsconfig.json

```diff
@@ -1,13 +1,28 @@
 {
     "compilerOptions": {
+      "declaration": true,
```

Comment:
storybook is now responsible for being able to build nimble-components and spright-components TypeScript source instead of just using build output so the tsconfig needs to align so there  is not weird behavior differences between the two during TS build.

I did not want to make sharing the TS config part of this change. I also want to see if switching to relying on the source actually fixes the turbosnap issues before leaning in harder on this pattern.

Reply:
Can you make sure we have a follow up task to track sharing config? Or add a note to #2118 and keep it open after this PR goes in? It seems highly likely they'll accidentally diverge over time.

Reply:
Created https://github.com/ni/nimble/issues/2133